internal/vanilla/vanilla.go

```diff
@@ -0,0 +1,181 @@
+package vanilla
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/go-sql-driver/mysql"
+)
+
+type Task struct {
+	Id        int64
+	Name      string
+	Completed bool
+}
+
+type VanillaRepository struct {
```

Comment:
Make this struct privated. To enforce the knowledge only of the interface!

Secondly, remove the sufix `Repository`, as the package must be named like so.

```suggestion
type vanilla struct {
```